lms/djangoapps/courseware/courses.py

```diff
@@ -415,3 +416,32 @@ def get_studio_url(course, page):
     if is_studio_course and is_mongo_course:
         studio_link = get_cms_course_link(course, page)
     return studio_link
+
+
+def get_entrance_exam_modules(request, course, student):
+    """
+    This returns all problems in an entrance exam
+    """
+
+    exam_modules = []
+    if settings.FEATURES.get('ENTRANCE_EXAMS', False):
+        if getattr(course, 'entrance_exam_enabled', False):
+            exam_key = UsageKey.from_string(course.entrance_exam_id)
+            exam_descriptor = modulestore().get_item(exam_key)
```

Comment:
I believe there will be a Mongo performance boost, if you pass in a depth=3 argument here as it will optimize round trips to the database to fetch all children for the current node.